src/pmi/pmi.cs

```diff
@@ -21,6 +22,23 @@ public static string MapPathToFileName(string path)
     }
 }
 
+class AsyncHelper
```

Comment:
It would be useful to have a header comment here explaining what these special-case types are for.